### PR TITLE
Use snprintf() instead of sprintf()

### PR DESCRIPTION
**Pull request by mywave82**
Compilers are starting to complain about sprintf usage.

### CODE REVIEW

File: src/d00.cpp [hunk 1]
```diff
@@ -509,8 +509,7 @@ void Cd00Player::rewind(int subsong)
 std::string Cd00Player::gettype()
 {
   char	tmpstr[40];
-
-  sprintf(tmpstr,"EdLib packed (version %d)",version > 1 ? header->version : header1->version);
+  snprintf(tmpstr, sizeof (tmpstr), "EdLib packed (version %d)",version > 1 ? header->version : header1->version);
```
Comment (binarymaster) [inline src/d00.cpp:512]:
```suggestion
  snprintf(tmpstr, sizeof (tmpstr), "EdLib packed (version %d)", version > 1 ? header->version : header1->version);
```

File: src/herad.cpp [hunk 2]
```diff
@@ -69,9 +69,9 @@ std::string CheradPlayer::gettype()
 {
 	char scomp[12 + 1] = "";
 	if (comp > HERAD_COMP_NONE)
-		sprintf(scomp, ", %s packed", (comp == HERAD_COMP_HSQ ? "HSQ" : "SQX"));
-	char type[40 + 1];
-	sprintf(type, "HERAD System %s (version %d%s)", (AGD ? "AGD" : "SDB"), (v2 ? 2 : 1), scomp);
+		snprintf(scomp, sizeof(scomp), ", %s packed", (comp == HERAD_COMP_HSQ ? "HSQ" : "SQX"));
+	char type[41 + 1];
+	snprintf(type, sizeof (type), "HERAD System %s (version %d%s)", (AGD ? "AGD" : "SDB"), (v2 ? 2 : 1), scomp);
```
Comment (binarymaster) [inline src/herad.cpp:74]:
```suggestion
	snprintf(type, sizeof(type), "HERAD System %s (version %d%s)", (AGD ? "AGD" : "SDB"), (v2 ? 2 : 1), scomp);
```

File: src/herad.cpp [hunk 3]
```diff
@@ -69,9 +69,9 @@ std::string CheradPlayer::gettype()
 {
 	char scomp[12 + 1] = "";
 	if (comp > HERAD_COMP_NONE)
-		sprintf(scomp, ", %s packed", (comp == HERAD_COMP_HSQ ? "HSQ" : "SQX"));
-	char type[40 + 1];
-	sprintf(type, "HERAD System %s (version %d%s)", (AGD ? "AGD" : "SDB"), (v2 ? 2 : 1), scomp);
+		snprintf(scomp, sizeof(scomp), ", %s packed", (comp == HERAD_COMP_HSQ ? "HSQ" : "SQX"));
+	char type[41 + 1];
```
Comment (binarymaster) [inline src/herad.cpp:73]: 40 + 1 is already the size with terminating zero, no need to change that.

File: src/msc.cpp [hunk 4]
```diff
@@ -176,7 +176,7 @@ std::string CmscPlayer::gettype()
 {
   char vstr [40];
 
-  sprintf(vstr, "AdLib MSCplay (version %d)", version);
+  snprintf(vstr, sizeof (vstr), "AdLib MSCplay (version %d)", version);
```
Comment (binarymaster) [inline src/msc.cpp:179]:
```suggestion
  snprintf(vstr, sizeof(vstr), "AdLib MSCplay (version %d)", version);
```

File: src/herad.cpp (hunk 3 above)
Comment (mywave82) [reply src/herad.cpp:73]:
```
 1234567890123456789012345678901234567890
"HERAD System AGD (version 2, HSQ packed)"
```
Yes you are right, 40 + 1 is enough.

**Comment by mywave82** [comment]
It is a bit messy @binarymaster that the different files have different coding styles regarding indenting, placement of space etc.

**Comment by binarymaster** [comment]
Yeah it's true, maybe we would need to apply standard formatting at some point.

**Pull request merged**